include/configuration/Heuristic.hpp

```diff
@@ -0,0 +1,158 @@
+//
+// This file is part of the MQT QMAP library released under the MIT license.
+// See README.md or go to https://github.com/cda-tum/qmap for more information.
+//
+
+#pragma once
+
+#include <iostream>
+
+enum class Heuristic {
+  /** maximum over all distances between any virtual qubit pair in the current
+     layer; optimizing gate-count; admissible; tight */
+  GateCountMaxDistance,
+  /** sum over all distances between any virtual qubit pair in the current
+     layer; optimizing gate-count; not admissible; tight */
+  GateCountSumDistance,
+  /** sum over all distances between any virtual qubit pair in the current layer
+     minus the upper limit of viable shared swaps; optimizing gate-count;
+     admissible; tight */
+  GateCountSumDistanceMinusSharedSwaps,
+  /** maximum of `Heuristic::GateCountMaxDistance` and
+     `Heuristic::GateCountSumDistanceMinusSharedSwaps`; optimizing gate-count;
+     admissible; tight */
+  GateCountMaxDistanceOrSumDistanceMinusSharedSwaps,
+  /** minimum cost if each virtual qubit pair/qubit is mapped to its
+     individually best physical edge/qubit; optimizing fidelity; admissible;
+     tight */
+  FidelityBestLocation
+};
+
+/**
+ * A heuristic is admissible if it never overestimates the cost of the best
+ * reachable goal node, i.e. c(n*) <= c(n) + h(n) for cost function c,
+ * heuristic h, any node n in the search graph, and n* the best reachable goal
+ * node from n.
+ */
+[[maybe_unused]] static inline bool isAdmissible(const Heuristic heuristic) {
```

Comment:
The mapper itself currently only needs to know if the heuristic is principally admissible. `isAdmissible` is therefore only used in the tests (which would result in a compiler warning, if the project is compiled without the tests). I thought about removing the function, but came to the conclusion that it's not unlikely that we come up with optimizations specifically for generally admissible heuristics further down the line. So its probably best to keep that information on all our heuristics, especially since its basically just 1 switch statement.
Also it is just a great sanity check in the tests. I caught many bugs in the current implementation only because I started to test for general admissibility.

Reply:
I see. Makes sense and I am totally on board with keeping the function 👍🏼